src/LiveComponent/src/EventListener/LiveComponentSubscriber.php

```diff
@@ -125,9 +125,11 @@ public function onKernelController(ControllerEvent $event): void
         if ($request->query->has('data')) {
             // ?data=
             $data = json_decode($request->query->get('data'), true, 512, \JSON_THROW_ON_ERROR);
-        } else {
+        } elseif ($request->request->has('data')) {
             // OR body of the request is JSON
-            $data = json_decode($request->getContent(), true, 512, \JSON_THROW_ON_ERROR);
+            $data = json_decode($request->request->get('data'), true, 512, \JSON_THROW_ON_ERROR);
+        } else {
+            $data = $request->query->all();
```

Comment:
For `else`, I would actually throw a `BadRequestHttpException` with a message, which will trigger a 400 error.

Reply:
I'll be interested to see how you implement handling the file uploads :). Are you planning to pass these are arguments to the `LiveAction`? That's what I had in mind - e.g.

```php
public function handleFileUpload(UploadedFile $myFile)
```

Where, I guess `$myFile` would need to match up with the `name` of your file input on the frontend. Though, what do we do then if the file input is `<input type="file" name="form[myFile]">`? :)

Reply:
I've added exception as suggested. I don't remember exactly why I've put it there, possibly some tests or live demo failures were present when that fallback was removed at the time.

About handling file uploads - my initial though was "I can just autowire `Request` and do whatever I want with files, but your comment sparked an idea that I'm exploring. More in the general comment as I will definitely need some hints on that :)